### PR TITLE
SIGINT-1111: Jenkins Plugin: Map platform specific bridge download URL from UI to code

### PR DESCRIPTION
**Pull request by maksudur-rahman-maruf**
-  Mapped platform specific bridge download URL from UI to code

### CODE REVIEW

**Review by jahid1209 — APPROVED** [review]

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 1]
```diff
@@ -42,17 +42,24 @@ public class ScannerGlobalConfig extends GlobalConfiguration implements Serializ
...
     private String coverityInstallationPath;
 
-    @HelpMarkdown("Provide the Synopsys Bridge artifactory URL from where synopsys-bridge zip file can be downloaded. " +
+    @HelpMarkdown("Provide the URL from where synopsys-bridge zip file can be downloaded. " +
         "Note: You must need to provide the full download url which includes the zip file path.")
     private String synopsysBridgeDownloadUrl;
 
-    @HelpMarkdown("Provide the specific Synopsys bridge version you want to download from artifactory.")
+
+    private String synopsysBridgeDownloadUrlForMac;
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:50]: add `HelpMarkdown`

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 2]
```diff
@@ -42,17 +42,24 @@ public class ScannerGlobalConfig extends GlobalConfiguration implements Serializ
...
+
+    private String synopsysBridgeDownloadUrlForWindows;
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:52]: add `HelpMarkdown`

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 3]
```diff
@@ -42,17 +42,24 @@ public class ScannerGlobalConfig extends GlobalConfiguration implements Serializ
...
+
+    private String synopsysBridgeDownloadUrlForLinux;
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:54]: add `HelpMarkdown`

File: src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java [hunk 4]
```diff
@@ -237,4 +242,14 @@ private static Map<String, Object> preparePolarisParametersMap(SecurityScanStep
...
     }
 
+    private static String getSynopsysBridgeDownloadUrlBasedOnAgentOS(FilePath workspace, TaskListener listener,
+                                                                     String synopsysBridgeDownloadUrlForMac,
+                                                                     String synopsysBridgeDownloadUrlForLinux,
+                                                                     String synopsysBridgeDownloadUrlForWindows) {
+        String agentOs = Utility.getAgentOs(workspace, listener);
+        if(agentOs.contains("mac")) return synopsysBridgeDownloadUrlForMac;
+        else if(agentOs.contains("linux")) return synopsysBridgeDownloadUrlForLinux;
+        else return synopsysBridgeDownloadUrlForWindows;
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java:252]: add curly braces.

format these as- 

```
if () {
} else if () {
}
```

**Review by zaman-akib — CHANGES_REQUESTED** [review]

File: src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java [hunk 5]
```diff
@@ -42,17 +42,25 @@ public class ScannerGlobalConfig extends GlobalConfiguration implements Serializ
     @HelpMarkdown("Provide the Coverity installation path on the agent where it will be installed during the Jenkins build.")
     private String coverityInstallationPath;
 
-    @HelpMarkdown("Provide the Synopsys Bridge artifactory URL from where synopsys-bridge zip file can be downloaded. " +
-        "Note: You must need to provide the full download url which includes the zip file path.")
-    private String synopsysBridgeDownloadUrl;
+    @HelpMarkdown("Provide the URL from where synopsys-bridge zip file can be downloaded. " +
+            "Note: You must need to provide the full download url which includes the zip file path.")
+    private String synopsysBridgeDownloadUrlForMac;
```
Comment (jahid1209) [inline src/main/java/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig.java:48]: Remove this blank line.

**Review by jahid1209 — APPROVED** [review]

**Review by zaman-akib — APPROVED** [review]

**Review by kishorikumar — APPROVED** [review]

**Pull request merged**